### PR TITLE
[WIP] Implement conversions between `java.lang.{Float, Double}` and JString

### PR DESCRIPTION
**Pull request by t-hiroyoshi**
target: #160 
ref: #173

### CODE REVIEW

**Comment by densh** [comment]
Needs a rebase on top of latest master due to CI changes.

File: javalib/src/main/scala/java/lang/Double.scala [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
-  override def equals(that: Any): scala.Boolean = that match {
-    case that: Double =>
-      val a = doubleValue
-      val b = that.doubleValue
-      (a == b) || (Double.isNaN(a) && Double.isNaN(b))
-
-    case _ =>
-      false
-  }
+  def this(s: String) =
```
Comment (densh) [inline javalib/src/main/scala/java/lang/Double.scala]: `@inline`

File: javalib/src/main/scala/java/lang/Float.scala [hunk 2]
```diff
@@ -1,25 +1,38 @@
 package java.lang
 
+import scalanative.native._
+
 final class Float(val floatValue: scala.Float)
     extends Number
     with Comparable[Float] {
-  def this(s: String) = this(Float.parseFloat(s))
+  def this(s: String) =
```
Comment (densh) [inline javalib/src/main/scala/java/lang/Float.scala]: `@inline`

File: javalib/src/main/scala/java/lang/HexStringParser.scala [hunk 3]
```diff
@@ -0,0 +1,228 @@
+package java.lang
+
+import java.util.regex.Pattern
```
Comment (densh) [inline javalib/src/main/scala/java/lang/HexStringParser.scala]: Unfortunately we don't have `java.util.regex` and therefore can't use regular expressions here. All regular expressions need to be rewritten into handwritten parsers or recognisers.

File: javalib/src/main/scala/java/lang/NumberConverter.scala [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+    new String(formattedDecimal, 0, charPos)
+  }
+
+  // ToDo reimplement from c to scala
+  def bigIntDigitGeneratorInstImpl(f: scala.Long,
+                                   e: scala.Int,
+                                   isDenormalized: scala.Boolean,
+                                   mantissaIsZero: scala.Boolean,
+                                   p: scala.Int): Unit = ???
```
Comment (densh) [inline javalib/src/main/scala/java/lang/NumberConverter.scala]: Needs implementation.

File: javalib/src/main/scala/java/lang/NumberParser.scala [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+  private case class StringExponentPair(s: String,
+                                        e: scala.Int,
+                                        negative: scala.Boolean)
+
+  // ToDo reimplement from c to scala
+  @inline private def parseDblImpl(s: String, e: scala.Int): scala.Double =
+    ???
+  @inline private def parseFltImpl(s: String, e: scala.Int): scala.Float =
+    ???
```
Comment (densh) [inline javalib/src/main/scala/java/lang/NumberParser.scala]: Needs implementation.

File: javalib/src/main/scala/java/lang/NumberParser.scala [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+  private val DOUBLE_MIN_EXP = -324
+  private val DOUBLE_MAX_EXP = 308
+
+  private case class StringExponentPair(s: String,
+                                        e: scala.Int,
+                                        negative: scala.Boolean)
+
+  // ToDo reimplement from c to scala
+  @inline private def parseDblImpl(s: String, e: scala.Int): scala.Double =
+    ???
```
Comment (densh) [inline javalib/src/main/scala/java/lang/NumberParser.scala]: Needs implementation.

**Comment by densh** [comment]
@t-hiroyoshi Any updates on this front?

**Comment by t-hiroyoshi** [comment]
@densh I'm busy at my work this month... I'll resume this RP next week.

**Comment by densh** [comment]
Superseded by #262

**Comment by densh** [comment]
This turned to be 100-th PR we've closed so far since the project has been open sourced in May. 🎉

**Pull request closed without merging**